### PR TITLE
Events tests

### PR DESCRIPTION
**Pull request by brahmaroutu**
Refactoring the events tests to make them extensible.

### CODE REVIEW

File: integration-cli/docker_cli_events_test.go [hunk 1]
```diff
@@ -317,26 +317,7 @@ func TestEventsFilterContainerID(t *testing.T) {
...
-				t.Fatalf("second event should be start, not %#v", startEvent)
-			}
-		}
-		if len(events) == 3 {
-			dieEvent := strings.Fields(events[len(events)-1])
-			if dieEvent[len(dieEvent)-1] != "die" {
-				t.Fatalf("event should be die, not %#v", dieEvent)
-			}
-		}
+		checkEvents(t, events[:len(events)-1])
```
Comment (vishh) [inline integration-cli/docker_cli_events_test.go]: Looks like the test used to pass even with just one event. @LK4D4 can you think of any reasons?

Comment (LK4D4) [inline integration-cli/docker_cli_events_test.go]: Hmm, actually there is `Fatal` in checkEvents `if len(events) != 3`. Have no Idea how it can pass

Comment (vishh) [inline integration-cli/docker_cli_events_test.go]: OK. The refactoring seems to make sense.

**Comment by vishh** [comment]
LGTM

**Comment by brahmaroutu** [comment]
@vishh Thanks for the review This is done to make #11827 a bit easier.

**Comment by LK4D4** [comment]
LGTM

**Pull request merged**